applications/GeoMechanicsApplication/tests/cpp_tests/test_time_stepping.cpp

```diff
@@ -0,0 +1,154 @@
+    unsigned int NumberOfExecuteFinalizeSolutionStepCalls()   const {return mSolutionStepFinalizedCalls;}
+
+private:
+    unsigned int mSolutionStepInitializedCalls = 0;
+    unsigned int mSolutionStepFinalizedCalls   = 0;
+};
+
+class DummyStrategyWrapper : public StrategyWrapper
+{
+public:
+    explicit DummyStrategyWrapper::DummyStrategyWrapper(bool convergence_needed) :
+        mConvergenceNeeded ( convergence_needed ? TimeStepEndState::ConvergenceState::converged : TimeStepEndState::ConvergenceState::non_converged) {}
+    [[nodiscard]] TimeStepEndState::ConvergenceState GetConvergenceState(const TimeStepEndState& rEndState) override { return mConvergenceNeeded;};
+    [[nodiscard]] std::size_t                        GetNumberOfIterations()                                const override { return 4;};
+    [[nodiscard]] double                             GetEndTime()                                           const override { return 10.;};
+    void Initialize()             override {mSolverStrategyInitializeCalls += 1;}
+    void InitializeSolutionStep() override {mSolverStrategyInitializeSolutionStepCalls += 1;}
+    void Predict()                override {mSolverStrategyPredictCalls += 1;}
+    bool SolveSolutionStep()      override {mSolverStrategySolveSolutionsStepCalls += 1;
+                                            return true;}
+    void FinalizeSolutionStep()   override {mSolverStrategyFinalizeSolutionStepCalls += 1;}
+    unsigned int NumberOfSolverStrategyInitializeCalls()             const {return mSolverStrategyInitializeCalls;}
+    unsigned int NumberOfSolverStrategyInitializeSolutionStepCalls() const {return mSolverStrategyInitializeSolutionStepCalls;}
+    unsigned int NumberOfSolverStrategyPredictCalls()                const {return mSolverStrategyPredictCalls;}
+    unsigned int NumberOfSolverStrategySolveSolutionStepCalls()      const {return mSolverStrategySolveSolutionsStepCalls;}
+    unsigned int NumberOfSolverStrategyFinalizeSolutionStepCalls()   const {return mSolverStrategyFinalizeSolutionStepCalls;}
+
+private:
+    TimeStepEndState::ConvergenceState mConvergenceNeeded;
+    unsigned int mSolverStrategyInitializeCalls             = 0;
+    unsigned int mSolverStrategyInitializeSolutionStepCalls = 0;
+    unsigned int mSolverStrategyPredictCalls                = 0;
+    unsigned int mSolverStrategySolveSolutionsStepCalls     = 0;
+    unsigned int mSolverStrategyFinalizeSolutionStepCalls   = 0;
+};
+
+}
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(RunReturnsNonConvergedWhenStrategyDoesNotConverge, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto nonconverging_strategy = std::make_shared<DummyStrategyWrapper>(false);
+    executor.SetSolverStrategy(nonconverging_strategy);
+    const auto time = 0.0;
+    KRATOS_EXPECT_EQ(TimeStepEndState::ConvergenceState::non_converged, executor.Run(time).convergence_state);
```

Comment:
I think we can now simply this to `KRATOS_EXPECT_TRUE(executor.Run(time).NonConverged())` since the `TimeStepEndState` struct now is equipped with `Converged()` and `NonConverged()` queries.

Reply:
Done